src/llama-model.cpp

```diff
@@ -4203,12 +4276,26 @@ struct llm_build_llama : public llm_graph_context {
         // inp_pos - contains the positions
         ggml_tensor * inp_pos = build_inp_pos();
 
+        // temperature tuning
+        ggml_tensor * inp_attn_scale = nullptr;
+        if (arch == LLM_ARCH_LLAMA4) {
+            auto inp = std::make_unique<llm_graph_input_attn_temp>(n_pos_per_token(), hparams.n_attn_temp_floor_scale, hparams.f_attn_temp_scale);
+            inp_attn_scale = ggml_new_tensor_3d(ctx0, GGML_TYPE_F32, 1, 1, n_tokens*n_pos_per_token());
+            ggml_set_input(inp_attn_scale);
+            inp->attn_scale = inp_attn_scale;
+            res->add_input(std::move(inp));
+        }
+
```

Comment:
Move this in a helper function for consistency:

```suggestion
        // temperature tuning
        ggml_tensor * inp_attn_scale = build_inp_attn_scale();
```

Reply:
Fixed in https://github.com/ggml-org/llama.cpp/pull/12791/commits/af1968c3547c7970d68590eb582d94f3a1b84ac9